File: LeetCode-Java/Binary Tree Level Order Traversal.java
/**
 * Definition for binary tree
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public List<List<Integer>> levelOrder(TreeNode root) {
        List<List<Integer>> result = new ArrayList<List<Integer>>();
        LinkedList<TreeNode> nodeStack = new LinkedList<TreeNode>();
        if(root == null) return result;
        nodeStack.addLast(root);
        int count = 1;
        while(count > 0)
        {
            List<Integer> tempResult = new ArrayList<Integer>();
            int newCount = 0;
            while(count > 0)
            {
                TreeNode tempNode = nodeStack.pollFirst();
                count --;
                tempResult.add(tempNode.val);
                if(tempNode.left != null) 
                {
                    nodeStack.addLast(tempNode.left);
                    newCount ++;
                }
                if(tempNode.right != null)
                {
                    nodeStack.addLast(tempNode.right);
                    newCount ++;
                }
            }
            if(tempResult.size() > 0) result.add(tempResult);
            count = newCount;
        }
        return result;
    }
}

File: Leetcode-cpp/binary_tree_level_order_traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
        vector<vector<int>> result;
        if (root == nullptr) return result;
        queue<TreeNode*> record;
        record.push(root);
        int count = 1;
        while (!record.empty()) {
            vector<int> temp_result;
            int new_count = 0;
            while (count > 0) {
                TreeNode* temp_node = record.front();
                count--;
                record.pop();
                temp_result.push_back(temp_node->val);
                if (temp_node->left != nullptr) {
                    new_count++;
                    record.push(temp_node->left);
                }
                if (temp_node->right != nullptr) {
                    new_count++;
                    record.push(temp_node->right);
                }
            }
            result.push_back(temp_result);
            count = new_count;
        }
        return result;
    }
};